qiime2/sdk/usage.py

```diff
@@ -922,6 +922,131 @@ def init_format(self, name: str,
+
+        Parameters
+        ----------
+        name : str
+            The canonical name of the variable to be returned.
+        url : str
+            The url of the Artifact that should be downloaded for the
+            example. If a QIIME 2 epoch (e.g., 2022.11) is part of the URL, as
+            might be the case if obtaining an Artifact from docs.qiime2.org,
+            it can be templated in by including `{qiime2.__release__}` in an
+            F-string defining the URL.
+
+        Returns
+        -------
+        UsageVariable
+            This particular return class can be changed by a driver which
+            overrides :meth:`usage_variable`.
+        """
+        # The following example needs to use an Artifact that the test suite's
+        # plugin manager can handle.
+        # Examples
+        # --------
+        # >>> import qiime2
+        # >>> url = 'https://data.qiime2.org/{qiime2.__release__}/data/tutorials/moving-pictures/table.qza' # noqa: E501
+        # >>> mvp_table = use.init_artifact_from_url('mvp_table', url)
+        # >>> mvp_table
+        # <ExecutionUsageVariable name='mvp_table', var_type='artifact'>
+        def factory():
+            import tempfile
+            import qiime2
+
+            data = self._request_url(url)
+
+            with tempfile.NamedTemporaryFile() as f:
+                f.write(data.read())
+                f.flush()
+                try:
+                    result = qiime2.Artifact.load(f.name)
+                except ValueError as ex:
+                    raise ValueError(
+                        'Could not load Artifact from URL data: %s\n'
+                        ' Original exception: %s'
+                        % (url, str(ex)))
+
+            return result
+
+        return self.init_artifact(name, factory)
+
+    def init_metadata_from_url(self, name: str, url: str, column: str = None,
+                               ) -> UsageVariable:
+        """Obtain metadata from a url.
+
+        Driver implementations may use this to intialize example metadata.
+
+        Parameters
+        ----------
+        name : str
+            The canonical name of the variable to be returned.
+        url : str
+            The url of the Artifact that should be downloaded for the
+            example. If a QIIME 2 epoch (e.g., 2022.11) is part of the URL, as
+            might be the case if obtaining an Artifact from docs.qiime2.org,
+            it can be templated in by including `{qiime2.__release__}` in an
+            F-string defining the URL (see the doc string for this method for
+            an example).
+
+        Returns
+        -------
+        UsageVariable
+            This particular return class can be changed by a driver which
+            overrides :meth:`usage_variable`.
+
+        Examples
+        --------
+        >>> import qiime2
+        >>> url = f'https://data.qiime2.org/{qiime2.__release__}/tutorials/moving-pictures/sample_metadata.tsv' # noqa: E501
```

Comment:
I can't figure out how to get a multi-line string working as input in a doc string. As it stands, I'm disabling the max line length check for flake8. Open to other ideas though.

Reply:
I think this will do what you want. Assuming I remembered how to show continued expressions in doctest.

```suggestion
        >>> url = (f'https://data.qiime2.org/{qiime2.__release__}/tutorials/'
        ...        'moving-pictures/sample_metadata.tsv')
```

Reply:
I tested @ebolyen's suggestion locally, and that worked for me!

Reply:
I could have sworn I tried that but I guess not! Or maybe something unrelated was failing when I tried and mistook it for this failing. In any case, thank you both!